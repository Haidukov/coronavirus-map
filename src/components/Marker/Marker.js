import React from 'react';

import styles from './Marker.module.css';

const Marker = ({ country, onClick, bigCircleSize, smallCircleSize }) => {
    return (
        <div 
            data-testid='marker'
            onClick={onClick}
        >
            <div
                data-testid='bigCircle'
                className={styles.bigCircle}
                style={{
                    position: 'relative',
                    top: -bigCircleSize / 2 + 'px',
                    left: -bigCircleSize / 2 + 'px',
                    width: bigCircleSize + 'px',
                    height: bigCircleSize + 'px',
                }}
            >
                <div
                    data-testid='smallCircle'
                    className={styles.smallCircle}
                    style={{
                        width: smallCircleSize + 'px',
                        height: smallCircleSize + 'px',
                    }}
                >
                </div>
            </div>
        </div>
    )
};

export default Marker;