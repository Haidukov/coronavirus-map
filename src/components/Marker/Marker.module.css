.bigCircle {
    display: flex;
    justify-content: center;
    align-items: center;
    background: red;
    border-radius: 50%;
    border: '.3px solid black'
}

.smallCircle {
    background: black;
    border-radius: 50%;
}