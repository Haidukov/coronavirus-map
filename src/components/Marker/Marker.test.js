import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Marker from './Marker';

describe('<Marker/>', () => {
    it('should match snapshot', () => {
        const wrapper = render(<Marker />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render data', () => {
        const { getByTestId, container } = render((
            <Marker 
                smallCircleSize={10} 
                bigCircleSize={100}
            />
        ));
        const bigCircle = getByTestId('bigCircle');
        const smallCircle = getByTestId('smallCircle');

        expect(bigCircle.style.top).toBe('-50px');
        expect(bigCircle.style.left).toBe('-50px');
        expect(bigCircle.style.height).toBe('100px');
        expect(bigCircle.style.width).toBe('100px');

        expect(smallCircle.style.height).toBe('10px');
        expect(smallCircle.style.width).toBe('10px');
        expect(container).toMatchSnapshot();
    });  

    it('should handle onclick event', () => {
        const onClick = jest.fn();
        const { getByTestId } = render(<Marker onClick={onClick}/>);
        const marker = getByTestId('marker');
        fireEvent.click(marker);
        expect(onClick).toHaveBeenCalled();
    });
});