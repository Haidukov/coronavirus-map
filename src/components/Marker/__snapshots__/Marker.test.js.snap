// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Marker/> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="marker"
      >
        <div
          class="bigCircle"
          data-testid="bigCircle"
          style="position: relative;"
        >
          <div
            class="smallCircle"
            data-testid="smallCircle"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="marker"
    >
      <div
        class="bigCircle"
        data-testid="bigCircle"
        style="position: relative;"
      >
        <div
          class="smallCircle"
          data-testid="smallCircle"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Marker/> should render data 1`] = `
<div>
  <div
    data-testid="marker"
  >
    <div
      class="bigCircle"
      data-testid="bigCircle"
      style="position: relative; top: -50px; left: -50px; width: 100px; height: 100px;"
    >
      <div
        class="smallCircle"
        data-testid="smallCircle"
        style="width: 10px; height: 10px;"
      />
    </div>
  </div>
</div>
`;
